src/uu/head/src/head.rs

```diff
@@ -522,15 +526,35 @@ fn uu_head(options: &HeadOptions) -> UResult<()> {
                     }
                     println!("==> {name} <==");
                 }
-                head_file(&mut file, options)?;
+                head_file(&mut file_handle, options)?;
+                Ok(())
+            }
+            None => {
+                // Handle files with non-UTF-8 names
```

Comment:
Way too much duplicated code. I think you can get around this by just doing a `if (file == "-")` above.